### ISSUE TITLE
OpenGL fullScreen opens on wrong monitor

### ISSUE THREAD

**Issue by neilcsmith-net** [processing/processing#3797]
In a multi-screen system, the JOGL Surface doesn't seem to respect the display passed into fullScreen correctly (on Linux at least).  I think the cause is -

https://github.com/processing/processing/blob/master/core/src/processing/opengl/PSurfaceJOGL.java#L348

GLWindow setFullscreen(boolean ) says that it uses the main monitor (which on my system seems somewhat random as they're the same size).  Should this line be using the call to setFullscreen(List<MonitorDevice> ) with a singleton list?

**Comment by codeanticode** [comment]
I didn't notice any problem with fullscreen when testing on Linux with multiple monitor config, but I will take another look. You are using beta 6, right?

**Comment by neilcsmith-net** [comment]
Yes, I updated today and tested before reporting. It's really odd that this
doesn't behave consistently the same way. I can't see where you're actually
setting the device to use if you're not passing in to setFullscreen? Top
level position is documented not to affect full screen windows?

Best wishes, Neil

Neil C Smith
Artist : Technologist : Adviser
http://neilcsmith.net

I didn't notice any problem with fullscreen when testing on Linux with
multiple monitor config, but I will take another look. You are using beta
6, right?

—
Reply to this email directly or view it on GitHub
<https://github.com/processing/processing/issues/3797#issuecomment-140208836>
.

**Comment by neilcsmith-net** [comment]
OK, tried removing the setTopLevelPosition() call from within if (fullScreen) and changing the following line to -

```java
window.setFullscreen(Collections.singletonList(displayDevice));
```

and this seems to be working consistently.  Looking at the source in JOGL it seems like the set position call checks whether the window is full screen, but at the time you call it it isn't, and it gets through.  Perhaps some combination of drivers / OS / screens take notice of the position and some don't?  I think the above more closely follows the spec, but maybe one to check on the JogAmp forum?  Certainly from there I'm not the only one to see / be confused by this issue.

**Comment by codeanticode** [comment]
Thanks for looking into this. Did you tested you fix on Windows and/or Mac?


On Tue, Sep 15, 2015, 2:58 PM Neil C Smith <notifications@github.com> wrote:

OK, tried removing the setTopLevelPosition() call from within if
(fullScreen) and changing the following line to -

window.setFullscreen(Collections.singletonList(displayDevice));

and this seems to be working consistently. Looking at the source in JOGL it
seems like the set position call checks whether the window is full screen,
but at the time you call it it isn't, and it gets through. Perhaps some
combination of drivers / OS / screens take notice of the position and some
don't? I think the above more closely follows the spec, but maybe one to
check on the JogAmp forum? Certainly from there I'm not the only one to see
/ be confused by this issue.

—
Reply to this email directly or view it on GitHub
<https://github.com/processing/processing/issues/3797#issuecomment-140500948>.<img
src="
https://ci3.googleusercontent.com/proxy/ZPHuc1SutquwDUkipoxcbU1LgruNE3Qt9-GePVC6yl3HXWhwg_RmAyBCXJ4DonWv-xBR6RPavbHmBM8L_SIKjG3UTeQmrXaQk9x_4kC0Tewbhm2uxqyZtpOo2NAxUJWgOtGTujiJKumBu4_S8_vlrB4uOpRUiQ=s0-d-e1-ft#https://github.com/notifications/beacon/AADzJqLGmHCdnBvqfNLsuMoaap3L8N5Cks5oyGHGgaJpZM4F9Dwh.gif
">

**Comment by codeanticode** [comment]
@JakubValtar I replaced ```window.setTopLevelPosition(sketchX, sketchY)``` with ```window.setFullscreen(Collections.singletonList(displayDevice))``` in [this line](https://github.com/processing/processing/blob/master/core/src/processing/opengl/PSurfaceJOGL.java#L341), as suggested by @neilcsmith-net. I tested the change on Mac, seems to work fine, but I don't have access to a multiple monitor right now, nor a windows machine. Could you take a look?

**Comment by JakubValtar** [comment]
Implemented in #3892
